### ISSUE TITLE
Autodetect highlight mode after save document

### ISSUE THREAD

**Issue by patricktokeeffe** [mate-desktop/pluma#529]
#### ~~Expected~~ Proposed behaviour

When saving (both new & edits), use existing functions to identify and apply the appropriate Highlight Mode.

#### Actual behaviour

The appropriate Highlight Mode is only selected when an existing file is opened. Saving a file that has a corresponding Highlight Mode does not automatically enable that Highlight Mode. 

#### Steps to reproduce the behaviour

\*

#### MATE general version

MATE 1.20.1
```
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.4 LTS
Release:	18.04
Codename:	bionic
```

#### Package version

```
pluma:
  Installed: 1.20.1-3ubuntu1
  Candidate: 1.20.1-3ubuntu1
  Version table:
 *** 1.20.1-3ubuntu1 500
        500 http://ports.ubuntu.com bionic/universe armhf Packages
        100 /var/lib/dpkg/status
```

#### Linux Distribution

`Linux picarro-nh3 4.15.0-1055-raspi2 #59-Ubuntu SMP PREEMPT Mon Feb 3 20:26:01 UTC 2020 armv7l armv7l armv7l GNU/Linux`

**Comment by cwendling** [comment]
This should most probably apply *only* if the user didn't select a mode manually (and none is already applied, probably).  If a user is editing a file that is detected as something else than what she wants, it'd be a huge PITA to have to re-choose after each save.  Agreed it's a corner case as detection should be quite good, but still.

**Comment by patricktokeeffe** [comment]
I agree, those are good points. Maybe only do detection during a new file save, but not when saving existing file?


On March 5, 2020 12:47:56 AM PST, Colomban Wendling <notifications@github.com> wrote:
>This should most probably apply *only* if the user didn't select a mode manually (and none is already applied, probably).  If a user is editing a file that is detected as something else than what she wants, it'd be a huge PITA to have to re-choose after each save.  Agreed it's a corner case as detection should be quite good, but still.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/mate-desktop/pluma/issues/529#issuecomment-595102863
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
